packages/multichain/src/scope/assert.ts

```diff
@@ -0,0 +1,71 @@
+import { JsonRpcError } from '@metamask/rpc-errors';
+import type { Hex } from '@metamask/utils';
+
+import {
+  isSupportedMethod,
+  isSupportedNotification,
+  isSupportedScopeString,
+} from './supported';
+import type { ScopeObject, ScopesObject } from './types';
+
+export const assertScopeSupported = (
+  scopeString: string,
+  scopeObject: ScopeObject,
```

Comment:
Unlike the removeScope() and removeAccount(), we're not applying a change to the ScopeObject. The ordering of scopeString first and then scopeObject is because a Scope (i.e. an entry in a ScopesObject), looks like:
```
{
  [scopeString]: scopeObject
}
```

open to suggestions. I felt it would be too weird to pass in an ScopesObject with a single entry